src/loader/AMASSLoader.py

```diff
@@ -0,0 +1,320 @@
+
+    {
+        "module": "loader.AMASSLoader",
+        "config": {
+        "data_path": "<args:0>",
+        "sub_dataset_id": "CMU",
+        "body_model_gender": "male",
+        "subject_id": "10",
+        "sequence_id": "1",
+        "frame_id": "600",
+        "cf_set_shading": "SMOOTH"
+      },
+    }
+
+    Example 2: generate a pose of human picking up golf ball (female model). from CMU dataset, choose subject number 64 which contains multiple trials of human playing golf, when frame_id is not specified, random frame will be chosen from the motion trial
+
+    {
+        "module": "loader.AMASSLoader",
+        "config": {
+        "data_path": "<args:0>",
+        "sub_dataset_id": "CMU",
+        "body_model_gender": "female",
+        "subject_id": "64",
+        "sequence_id": "27",
+        "frame_id": "",
+        "cf_set_shading": "SMOOTH"
+      },
+    }
+
+    **Configuration**:
+
+    .. csv-table::
+       :header: "Parameter", "Description"
+
+       "data_path", "The path to the AMASS Dataset folder in resources folder. Type: string. Default: 'resources/AMASS'."
+       "_used_sub_dataset_id", "Identifier for the sub dataset, the dataset which the human pose object should be extracted from. Type: srtring." 
+       "_used_body_model_gender", "The model gender, pose will represented using male, female or neutral body shape. Type: string." 
+       "_used_subject_id", "Type of motion from which the pose should be extracted, this is dataset dependent parameter. Type: int." 
+       "_used_sequence_id", "Sequence id in the dataset, sequences are the motion recorded to represent certain action. Type: int." 
+       "_used_frame_id", "Frame id in a selected motion sequence. Type: int."
+    """
+
+    # list of all possible supported mocap datasets: ['CMU', 'Transitions_mocap', 'MPI_Limits', 'SSM_synced',
+    # 'TotalCapture', 'Eyes_Japan_Dataset', 'MPI_mosh', 'MPI_HDM05', 'HumanEva', 'ACCAD', 'EKUT', 'SFU', 'KIT',
+    # 'H36M', 'TCD_handMocap', 'BML']
+
+    # dictionary contains mocap dataset name and path to its sub folder within the main dataset, dictionary will
+    # be filled from taxonomy.json file which indicates the supported datastests
+    supported_mocap_datasets = {}
+
+    # hex values for human skin tone to sample from
+    human_skin_colors = ['2D221E', '3C2E28', '4B3932', '5A453C', '695046', '785C50', '87675A', '967264', 'A57E6E',
+                         'B48A78', 'C39582', 'D2A18C', 'E1AC96', 'F0B8A0', 'FFC3AA', 'FFCEB4', 'FFDABE', 'FFE5C8']
+
+    def __init__(self, config):
+        LoaderInterface.__init__(self, config)
+        self._data_path = Utility.resolve_path(
+            self.config.get_string("data_path", "resources/AMASS"))
+        # Body Model Specs
+        self._used_body_model_gender = self.config.get_string(
+            "body_model_gender")
+        self._num_betas = 10  # number of body parameters
+        self._num_dmpls = 8  # number of DMPL parameters
```

Comment:
Are these numbers fixed forever? Or where do they come from?

Reply:
These numbers are based on a recommendation from the authors. refer to visualization tutorial from the authors: https://github.com/nghorbani/amass/blob/master/notebooks/01-AMASS_Visualization.ipynb